Add Gitbook migration stub (#2916)
withastro/docs: src/content/docs/en/guides/migrate-to-astro/from-gitbook.mdx

  <PackageManagerTabs>
    <Fragment slot="npm">
    ```shell
    npm create astro@latest -- --template docs
    ```
    </Fragment>
    <Fragment slot="pnpm">
    ```shell
    pnpm create astro@latest --template docs
    ```
    </Fragment>
    <Fragment slot="yarn">
    ```shell
    yarn create astro --template docs
    ```
    </Fragment>
  </PackageManagerTabs>

Once you have a new Astro project, you can sync your existing GitBook content to your new Astro project. GitBook has a [gitsync feature](https://docs.gitbook.com/product-tour/git-sync) that will automatically sync your GitBook content to a GitHub/GitLab repository. To sync directly to the docs template's content collection, specify `src/content/docs/en` or `src/content/docs` as the project directory.

:::caution
When enabling git sync be sure to specify "**GitBook to GitHub**" as the priority. This will ensure that your GitBook content is synced to your GitHub repository. Otherwise, you will overwrite your existing GitBook content.

> ![Choose Priority](/img/gitbook/gitbook-gitsync.png)

Reviewer: Does this file exist? Can't seem to find it anywhere

Author: haha, oh wait I think this got stashed somewhere when working on another branch. I'll see if I can find it now.

Author: found it!

> Once you have a new Astro project, you can sync your existing GitBook content to your new Astro project. GitBook has a [gitsync feature](https://docs.gitbook.com/product-tour/git-sync) that will automatically sync your GitBook content to a GitHub/GitLab repository. To sync directly to the docs template's content collection, specify `src/content/docs/en` or `src/content/docs` as the project directory.
> 
> :::caution
> When enabling git sync be sure to specify "**GitBook to GitHub**" as the priority. This will ensure that your GitBook content is synced to your GitHub repository. Otherwise, you will overwrite your existing GitBook content.
> ![Choose Priority](/guides/gitbook-gitsync.png)
> :::
> 
> After syncing the content to your repository and deleting the sync, you'll have a copy of your GitBook content in your repository. There are a few things to take into account before you can use this content in your Astro project:
> 
> - Gitbook uses a special syntax that differs from Markdoc where the `/` prefix denoting a closing tag is replaced with `end` for Gitbook files. In addition, some features of Gitbook rely on custom Markdoc components.
> - Astro's [Markdoc integration](/en/guides/integrations-guide/markdoc/) requires that the file extension be `.mdoc`. This is to avoid conflicts with other Markdown extensions like `.mdx` and `.md`.

Reviewer 2: ```suggestion
Once you have a new Astro project, you can sync your existing GitBook content to your new Astro project. GitBook has a [gitsync feature](https://docs.gitbook.com/product-tour/git-sync) that will automatically sync your GitBook content to a GitHub/GitLab repository. 

To sync directly to the docs template's content collection, specify `src/content/docs/en` or `src/content/docs` as the project directory.

:::caution
When enabling git sync be sure to specify "**GitBook to GitHub**" as the priority. This will ensure that your GitBook content is synced to your GitHub repository. Otherwise, you will overwrite your existing GitBook content.
:::

After syncing the content, you will now have a copy of your GitBook content in your Astro repository. Disable git sync to prevent future syncing with GitBook.

Note that although you now have your content migrated to your Astro project, it will not be immediately usable. To use this content in your Astro site, you will need to spend some time manually changing Gitbook's Markdoc into a format compatible with Astro. In particular:

- Astro's [Markdoc integration](/en/guides/integrations-guide/markdoc/) requires that the file extension be `.mdoc`. This is to avoid conflicts with other Markdown extensions like `.mdx` and `.md`.
- Gitbook uses a special syntax that differs from Markdoc where the `/` prefix denoting a closing tag is replaced with `end` for Gitbook files. You will need to update this notation throughout your files.
- Some features of Gitbook rely on custom Markdoc components. These components will not exist in Astro and must be deleted from your Markdoc code.
- The git sync only syncs your content, not any custom components you may have created. You will need to rebuild any components as Astro components, or UI framework components.
```
FAIR WARNING: The above is "Sarah AI" - may not actually be correct, but sounds good!!  :sweat_smile:

 "deleted the sync" sounds a little odd to me, so I just wrote placeholder content above with what I think is something like the correct wording.

Also, I wanted to emphasis "you're gonna have to do stuff!! This will not just work!!" so I added more to the intro to the points to consider. And, I changed the sense of "things to be aware of" to try to make them "things to do." In the process, I might have totally made up the last (new) point about components? :sweat_smile:  (I'm not sure whether people DO create their own components in GitBook??) but if so, that's the sort of thing we should be saying!  :smile:

Author: Sounds good to me! Custom components I don't believe are possible in GitBook at the moment, so I've left that final bullet point out, but this reads much better, thanks!

---
title: Migrating from GitBook
description: Tips for migrating an existing GitBook project to Astro
type: migration
stub: true
framework: GitBook
i18nReady: false
---

import PackageManagerTabs from '~/components/tabs/PackageManagerTabs.astro'

[GitBook](https://gitbook.com) is a web-based platform for creating and publishing documentation and books in a collaborative manner, with version control integration and customizable features.

## Key Similarities between GitBook and Astro

GitBook and Astro share some similarities that will help you migrate your project:

> - Both Astro and GitBook support [Markdown](/en/guides/markdown-content/). You can migrate all your existing documentation utilizing GitBook's gitsync feature.

Reviewer 3: For consistency, I would change "gitsync" to "Git Sync" so it is actually written the same way as in GitBook's docs.
```suggestion
- Both Astro and GitBook support [Markdown](/en/guides/markdown-content/). You can migrate all your existing documentation utilizing GitBook's Git Sync feature.
```

- Both Astro and GitBook use some form of [file-based routing](/en/core-concepts/routing/). Using Astro's file structure for your existing content and when adding new pages should feel familiar.

## Key Differences between GitBook and Astro

When you migrate your GitBook docs to Astro, you will notice some important differences:

> - A GitBook site is edited using an online dashboard. In Astro, you will use a [code editor](/en/editor-setup/) and development environment to maintain your site. You can develop locally on your machine, or choose a cloud editor/development environment like StackBlitz, CodeSandbox or Gitpod.

Reviewer 3: Missing serial comma
```suggestion
- A GitBook site is edited using an online dashboard. In Astro, you will use a [code editor](/en/editor-setup/) and development environment to maintain your site. You can develop locally on your machine, or choose a cloud editor/development environment like StackBlitz, CodeSandbox, or Gitpod.
```

> - GitBook stores your content in a database. In Astro, you will have individual files (typically Markdown or MDX) in your [project directory](/en/core-concepts/project-structure/) for each page's content. Or, you can choose to use a [CMS for your content](/en/guides/cms/), even your existing GitBook instance, and use Astro to fetch and present the data.

Reviewer 3: > even your existing GitBook instance, and use Astro to fetch and present the data.

So, from what we explored before, looks like GitBook doesn't actually give you content in its API (currently), so I'm not sure about wording this like you can use GitBook and just fetch to Astro. True, you could use scraping but I think only a few peeps would try to go this route.

If that's really the case, I would just remove this bit.
```suggestion
- GitBook stores your content in a database. In Astro, you will have individual files (typically Markdown or MDX) in your [project directory](/en/core-concepts/project-structure/) for each page's content. Or, you can choose to use a [CMS for your content](/en/guides/cms/) and use Astro to fetch and present the data.
```

> - GitBook sites use a specialised version of Markdoc (a superset of Markdown) for content. Astro also supports Markdoc via the optional [Markdoc integration](/en/guides/integrations-guide/markdoc/) to use `.mdx` files in addition to standard Markdown files.

Reviewer 3: Here, it should be `.mdoc` instead of `.mdx`, right? Also, we can't be sure if GitBook is using Markdoc itself, looks like they're using a custom solution closer to Markdoc and other tools like Jekyll. To not be misleading, what do you think of these changes?

```suggestion
- GitBook uses a custom syntax on top of Markdown for content. Astro supports Markdoc via the optional [Markdoc integration](/en/guides/integrations-guide/markdoc/), which features a similar syntax to the one you would use in GitBook.
```

## Switch from GitBook to Astro

To convert a GitBook documentation site to Astro, start with our official docs theme starter template, or explore more community docs themes in our [theme showcase](https://astro.build/themes?search=&categories%5B%5D=docs).

You can pass a `--template` argument to the `create astro` command to start a new Astro project with one of our official starters. Or, you can [start a new project from any existing Astro repository on GitHub](/en/install/auto/#starter-templates).

  <PackageManagerTabs>
    <Fragment slot="npm">
    ```shell
    npm create astro@latest -- --template docs
    ```
    </Fragment>
    <Fragment slot="pnpm">
    ```shell
    pnpm create astro@latest --template docs
    ```
    </Fragment>
    <Fragment slot="yarn">
    ```shell
    yarn create astro --template docs
    ```
    </Fragment>
  </PackageManagerTabs>

> Once you have a new Astro project, you can sync your existing GitBook content to your new Astro project. GitBook has a [gitsync feature](https://docs.gitbook.com/product-tour/git-sync) that will automatically sync your GitBook content to a GitHub/GitLab repository. 

Reviewer 3: ```suggestion
Once you have a new Astro project, you can sync your existing GitBook content to your new Astro project. GitBook has a [Git Sync feature](https://docs.gitbook.com/product-tour/git-sync) that will automatically sync your GitBook content to a GitHub/GitLab repository. 
```

To sync directly to the docs template's content collection, specify `src/content/docs/en` or `src/content/docs` as the project directory.

:::caution

> When enabling git sync be sure to specify "**GitBook to GitHub**" as the priority. This will ensure that your GitBook content is synced to your GitHub repository. Otherwise, you will overwrite your existing GitBook content.

Reviewer 3: ```suggestion
When enabling Git Sync be sure to specify "**GitBook to GitHub**" as the priority. This will ensure that your GitBook content is synced to your GitHub repository. Otherwise, you will overwrite your existing GitBook content.
```

:::

After syncing the content, you will now have a copy of your GitBook content in your Astro repository. Disable git sync to prevent future syncing with GitBook.
Note that although you now have your content migrated to your Astro project, it will not be immediately usable. To use this content in your Astro site, you will need to spend some time manually changing Gitbook's Markdoc into a format compatible with Astro. In particular:

- Astro's [Markdoc integration](/en/guides/integrations-guide/markdoc/) requires that the file extension be `.mdoc`. This is to avoid conflicts with other Markdown extensions like `.mdx` and `.md`.

> - Gitbook uses a special syntax that differs from Markdoc where the `/` prefix denoting a closing tag is replaced with `end` for Gitbook files. You will need to update this notation throughout your files.

Reviewer 3: Same here, to not be misleading.
```suggestion
- GitBook syntax differs from Markdoc where the `/` prefix denoting a closing tag is replaced with `end` for GitBook files. You will need to update this notation throughout your files.
```

> - Some features of Gitbook rely on custom Markdoc components. These components will not exist in Astro and must be deleted from your Markdoc code.

Reviewer 3: Here we should clarify the user has the option of actually creating these components after changing the tag syntax.
```suggestion
- Some features of GitBook rely on custom components. These components will not exist in Astro and must be created and added to your project through [Markdoc's config `tags` attribute](/en/guides/integrations-guide/markdoc/#render-markdoc-tags-as-astro-components) or removed from your files.
```
